services/service.go

```diff
@@ -416,13 +416,9 @@ func (s *Service) HandleQueryBroadcastFinished(recq *QueryBroadcastFinished) (ne
 
 // NewProtocol creates a protocol instance executed by all nodes
 func (s *Service) NewProtocol(tn *onet.TreeNodeInstance, conf *onet.GenericConfig) (onet.ProtocolInstance, error) {
-	err := tn.SetConfig(conf)
-	if err != nil {
-		return nil, err
-	}
+	tn.SetConfig(conf)
```

Comment:
Don't drop the error.

Reply:
If I don't drop the error 7f37465 some tests stop working
e.g. - `^TestServiceClearAttr$` in `github.com/ldsec/unlynx/services `

```
 - couldn't set config: Can't set config twice:
        go.dedis.ch/onet/v3.(*TreeNodeInstance).SetConfig
            /Users/jagomes/Go/pkg/mod/go.dedis.ch/onet/v3@v3.1.1/treenode.go:879:
```

Reply:
That might indicate a bug :P

`Service.NewProtocol` is handled specially by onet, to start new protocol on the children of the root node of the tree. To fix it, move the `SetConfig` in `StartProtocol`, just before calling `NewProtocol`, this way, only root will actually set the config for the whole tree, ensuring that every node get's it before starting.

Reply:
Ok this is perfect it works like a charm 6340084

Reply:
> That might indicate a bug :P
> 
> `Service.NewProtocol` is handled specially by onet, to start new protocol on the children of the root node of the tree. To fix it, move the `SetConfig` in `StartProtocol`, just before calling `NewProtocol`, this way, only root will actually set the config for the whole tree, ensuring that every node get's it before starting.

wow - @tharvik gets an onet expert! I know whom I'll ask for my next bug ;)